src/test/java/org/snomed/snowstorm/validation/DroolValidationServiceTest.java

```diff
@@ -0,0 +1,192 @@
+        assertEquals(4, invalidContents.size());
+
+        int index = 0;
+        assertEquals(Severity.WARNING, invalidContents.get(index).getSeverity());
+        assertEquals("Test resources were not available so assertions like case significance and US specific terms checks will not have run.", invalidContents.get(index).getMessage());
+
+        index++;
+        assertEquals(Severity.WARNING, invalidContents.get(index).getSeverity());
+        assertEquals("Active FSN should end with a valid semantic tag.", invalidContents.get(index).getMessage());
+
+        index++;
+        assertEquals(Severity.ERROR, invalidContents.get(index).getSeverity());
+        assertEquals("Active concepts must have at least one IS A relationship.", invalidContents.get(index).getMessage());
+
+        index++;
+        assertEquals(Severity.WARNING, invalidContents.get(index).getSeverity());
+        assertEquals("A concept's semantic tags should be compatible with those of the active parents.", invalidContents.get(index).getMessage());
+    }
+
+    @Test
+    void testValidateConceptAgainstConcreteValue() throws ServiceException {
+        Concept foundConcept = conceptService.find("100001", DEFAULT_BRANCH);
+        assertNotNull(foundConcept);
+
+        foundConcept.getDescriptions().stream().forEach(description -> description.setAcceptabilityMap(Collections.singletonMap(Concepts.US_EN_LANG_REFSET,
+                Concepts.descriptionAcceptabilityNames.get(PREFERRED))));
+
+        foundConcept.addRelationship(new Relationship("100002", ISA, SNOMEDCT_ROOT).setCharacteristicTypeId(Concepts.STATED_RELATIONSHIP));
+
+        ConcreteValue value = ConcreteValue.from("10", "int");
+        foundConcept.addRelationship(new Relationship("100003", "23131313", value));
+
+        final Concept updatedConcept = conceptService.update(foundConcept, DEFAULT_BRANCH);
+        assertEquals(2, updatedConcept.getRelationships().size());
+
+        List <InvalidContent> invalidContents = droolValidationService.validateConcept(DEFAULT_BRANCH, updatedConcept);
+        assertEquals(3, invalidContents.size());
+
+        int index = 0;
+        assertEquals(Severity.WARNING, invalidContents.get(index).getSeverity());
+        assertEquals("Test resources were not available so assertions like case significance and US specific terms checks will not have run.", invalidContents.get(index).getMessage());
+
+        index++;
+        assertEquals(Severity.WARNING, invalidContents.get(index).getSeverity());
+        assertEquals("Active FSN should end with a valid semantic tag.", invalidContents.get(index).getMessage());
+
+        index++;
+        assertEquals(Severity.WARNING, invalidContents.get(index).getSeverity());
+        assertEquals("A concept's semantic tags should be compatible with those of the active parents.", invalidContents.get(index).getMessage());
+    }
+
+    private void checkoutDroolsRules() throws GitAPIException, IOException {
```

Comment:
Very cool that you can check out another project using Java, however I have concerns with this approach. I would rather keep the repositories separate otherwise the Snowstorm build could fail when the content in the drools rules project changes.
Could you remove this git clone functionality and just clone two or three snomed drools rules into the snowstorm test resources instead please?
You could copy the rule you are using for FSN validation and a rule to validate relationships to check that doesn't break with concrete domains.

Reply:
All above-changed requests have been fixed. @kaicode